graphium/trainer/predictor.py

```diff
@@ -96,20 +95,46 @@ def __init__(
             )
             eval_options[task].check_metrics_validity()
 
+        # Work-around to retain task level when model_kwargs are modified for FullGraphFinetuningNetwork
+        if "task_heads_kwargs" in model_kwargs.keys():
+            task_heads_kwargs = model_kwargs["task_heads_kwargs"]
+        elif "pretrained_model_kwargs" in model_kwargs.keys():
+            # This covers finetuning cases where we finetune from the task_heads
+            task_heads_kwargs = model_kwargs["pretrained_model_kwargs"]["task_heads_kwargs"]
+        else:
+            raise ValueError("incorrect model_kwargs")
+        self.task_heads_kwargs = task_heads_kwargs
+
         self._eval_options_dict: Dict[str, EvalOptions] = eval_options
         self._eval_options_dict = {
             self._get_task_key(
-                task_level=model_kwargs["task_heads_kwargs"][key]["task_level"], task=key
+                task_level=task_heads_kwargs[key]["task_level"],
+                task=key
+                # task_level=model_kwargs["task_heads_kwargs"][key]["task_level"], task=key
```

Comment:
In general, the `PredictorModule` should be agnostic to the model passed. By having the `self._get_task_key` here, it forces a certain architecture in the config which is not very flexible. 

I see that this logic was introduced already in the code prior to this PR. If it requires too many changes, let's open a new issue.

Reply:
You are right, thanks for pointing out. 

We could achieve this by getting the task-specific information (which is only the task level as far as I know) from the datamodule.

Here is how this could be done:
a3d471540bc3f69cc0af2346ab0e66141b65311b

What do you think?